app/code/community/Bolt/Boltpay/Model/Observer.php

```diff
@@ -94,11 +94,21 @@ public function verifyOrderContents($observer)
                 $message .= "/transaction/$reference";
                 $message .= "<br/>Bolt reports ".($transaction->amount->amount/100).'. Magento expects '.$magentoTotal/100;
 
-                // TODO: Properly adjust amount if it is off by only one cent
-                $order->setHoldBeforeState($order->getState());
-                $order->setHoldBeforeStatus($order->getStatus());
-                $order->setState(Mage_Sales_Model_Order::STATE_HOLDED, true, $message)
-                    ->save();
+                # Adjust amount if it is off by only one cent, likely due to rounding
+                $difference = $transaction->amount->amount - $magentoTotal;
+                if ( abs($difference) <= 1) {
```

Comment:
I prefer to keep add one more condition to be explicit:

```
if ($difference == 0) {
  // no-op
} else if (abs($difference) == 1) {
  // adjust
} else {
  // on-hold
}
```

Reply:
@daisy1754 

Please check this code:
https://github.com/BoltApp/bolt-magento1/blob/92a5f7eeba9b7f2ee3bea9e16a54379d6afe6443/app/code/community/Bolt/Boltpay/Model/Observer.php#L90-L99

Do you want this even though the parent if block implicitly checks for this and therefore makes the case `if ($difference == 0)` to be impossible to ever be true?  I changed `if (abs($difference) == 1)` as you asked because the numbers should always be integers.